sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ParallelDocumentQueryExecutionContextBase.java

```diff
@@ -68,102 +66,105 @@ protected ParallelDocumentQueryExecutionContextBase(DiagnosticsClientContext dia
-        Utils.ValueHolder<Map<String, TContinuationToken>> outPartitionRangeToContinuation,
+    protected <TContinuationToken> FeedRangeEpkImpl findTargetRangeAndExtractContinuationTokens(
+        List<FeedRangeEpkImpl> feedRanges, Range<String> range,
+        Utils.ValueHolder<Map<FeedRangeEpkImpl, TContinuationToken>> outPartitionRangeToContinuation,
         TContinuationToken continuation) {
-        if (partitionKeyRanges == null) {
+        if (feedRanges == null) {
             throw new IllegalArgumentException("partitionKeyRanges can not be null.");
         }
 
-        if (partitionKeyRanges.size() < 1) {
+        if (feedRanges.size() < 1) {
             throw new IllegalArgumentException("partitionKeyRanges must have atleast one element.");
         }
 
-        for (PartitionKeyRange partitionKeyRange : partitionKeyRanges) {
-            if (partitionKeyRange == null) {
+        for (FeedRangeEpkImpl feedRangeEpk : feedRanges) {
+            if (feedRangeEpk == null) {
                 throw new IllegalArgumentException("partitionKeyRanges can not have null elements.");
             }
         }
 
         // Find the minimum index.
         PartitionKeyRange needle = new PartitionKeyRange(/* id */ null, range.getMin(), range.getMax());
+        FeedRangeEpkImpl feedRangeEpk = null;
         int minIndex;
-        for (minIndex = 0; minIndex < partitionKeyRanges.size(); minIndex++) {
-            if (needle.getMinInclusive().equals(partitionKeyRanges.get(minIndex).getMinInclusive())) {
+        for (minIndex = 0; minIndex < feedRanges.size(); minIndex++) {
+            feedRangeEpk = feedRanges.get(minIndex);
+            if (needle.getMinInclusive().equals(feedRanges.get(minIndex).getRange().getMin())) {
                 break;
             }
         }
 
-        if (minIndex == partitionKeyRanges.size()) {
+        if (minIndex == feedRanges.size()) {
             throw BridgeInternal.createCosmosException(HttpConstants.StatusCodes.BADREQUEST,
                     String.format("Could not find partition key range for continuation token: {0}", needle));
         }
 
-        List<PartitionKeyRange> replacementRanges;
+        List<FeedRangeEpkImpl> replacementRanges;
 
         // find what ranges make up the supplied continuation token
-        replacementRanges = partitionKeyRanges.stream()
-                                .filter(p -> range.getMin().compareTo(p.getMinInclusive()) <= 0 &&
-                                                 range.getMax().compareTo(p.getMaxExclusive()) >= 0)
-                                .sorted(Comparator.comparing(PartitionKeyRange::getId))
+        replacementRanges = feedRanges.stream()
+                                .filter(p -> range.getMin().compareTo(p.getRange().getMin()) <= 0 &&
+                                                 range.getMax().compareTo(p.getRange().getMax()) >= 0)
+//                                .sorted(Comparator.comparing(PartitionKeyRange::getId))
```

Comment:
we will need to sort. no?

is this intentional to remove the sorting part?

Reply:
This logic is not being used anymore. Logic for finding the partition for continuation is now moved to ```PartitionMapper``` class

Reply:
Also, I deleted this unused code